### PR TITLE
Run rustfmt

### PR DESCRIPTION
**Pull request by steveklabnik**
Please review carefully, as rustfmt is still under development and it might be a bit weird in places. We can fix anything that looks odd.

Includes https://github.com/brianloveswords/hookshot/pull/21 because it turns out it matters.

### CODE REVIEW

File: src/cli.rs [hunk 1]
```diff
@@ -36,7 +36,9 @@ pub fn main() {
     opts.optflag("h", "help", "print this help menu");
 
     let matches = match opts.parse(&args[1..]) {
-        Ok(m) => { m }
+        Ok(m) => {
+            m
+        }
```
Comment (steveklabnik) [inline src/cli.rs:41]: this might be worse

File: src/config.rs [hunk 2]
```diff
@@ -252,9 +247,11 @@ impl Config {
                 if !Path::new(path).is_absolute() {
                     return Err(ConfigError {
                         desc: "entries in `playbooks` must be absolute paths",
-                        detail: Some(format!("'{}.playbooks.{}' is not an absolute path", name, key)),
+                        detail: Some(format!("'{}.playbooks.{}' is not an absolute path",
+                                             name,
+                                             key)),
```
Comment (steveklabnik) [inline src/config.rs:252]: i wonder about this one too

File: src/config.rs [hunk 3]
```diff
@@ -425,7 +428,9 @@ mod tests {
             [app]
             [app.playbooks]
               a = "/path/to/playbook.yml"
-        "#.to_string()).unwrap();
+        "#
+                                        .to_string())
+                    .unwrap();
```
Comment (steveklabnik) [inline src/config.rs:433]: hmm, this might be very wrong.

Comment (steveklabnik) [inline src/config.rs:433]: and below

File: src/git.rs [hunk 4]
```diff
@@ -1,11 +1,12 @@
 //! A library for interacting with the git cli.
 //!
 //! This library provides an interface for operating on git repository. It is
-//! not intended to provide a 1-1 interface to the git cli but instead provide a
+// ! not intended to provide a 1-1 interface to the git cli but instead provide
+// a
```
Comment (steveklabnik) [inline src/git.rs:5]: this looks like a bug! filed https://github.com/nrc/rustfmt/issues/524

File: src/git.rs [hunk 5]
```diff
@@ -86,24 +88,24 @@ impl GitRepo {
...
+                               .current_dir(&self.local_path)
+                               .arg("fetch")
+                               .output() {
+            Ok(r) => r,
+            Err(e) => return Err(CommandError {
+                desc: "failed to execute process, see detail",
+                output: None,
+                detail: Some(format!("{}", e)),
+            }),
+        };
```
Comment (steveklabnik) [inline src/git.rs:100]: this is... interesting

File: src/notifier.rs [hunk 6]
```diff
@@ -30,11 +30,13 @@ enum TaskState {
...
 impl Display for TaskState {
     fn fmt(&self, f: &mut Formatter) -> fmt::Result {
-        write!(f, "{}", match *self {
-            TaskState::Started => "started",
-            TaskState::Success => "success",
-            TaskState::Failed => "failed",
-        })
+        write!(f,
+               "{}",
+               match *self {
```
Comment (steveklabnik) [inline src/notifier.rs:35]: hrm

File: src/task_manager.rs [hunk 7]
```diff
@@ -441,17 +480,43 @@ mod tests {
...
+                                     s: s.clone(),
+                                     m: "h",
+                                 });
+                manager.add_task(&queue_key,
+                                 Task {
+                                     s: s.clone(),
+                                     m: "s",
+                                 })
+                       .unwrap()
+                       .recv();
```
Comment (steveklabnik) [inline src/task_manager.rs:514]: yeowch

File: src/config.rs (hunk 3 above)
Comment (nrc) [inline src/config.rs:433]: There's a fix for these ones just landed - https://github.com/nrc/rustfmt/commit/2d267b16d1f28e7ab14cd1cc9953ae21fe1856c6 - maybe worth rebuilding and re-running

File: src/notifier.rs (hunk 6 above)
Comment (nrc) [inline src/notifier.rs:35]: https://github.com/nrc/rustfmt/issues/525

File: src/task_manager.rs (hunk 7 above)
Comment (marcusklaas) [inline src/task_manager.rs:514]: Yea.. not good. Try again with `struct_lit_width = 25`?

File: src/message.rs [hunk 8]
```diff
@@ -54,19 +54,21 @@ impl GitHubMessage {
...
-                (Some(b), Some(reftype), Some(branch))
-                    if *b == "refs"  && (*reftype == "heads" || *reftype == "tags") => branch,
-                _ => return Err("could not unpack `ref`")
-            }.to_string()
+                (Some(b), Some(reftype), Some(branch)) if *b == "refs" &&
+                                                          (*reftype == "heads" ||
+                                                           *reftype == "tags") => branch,
+                _ => return Err("could not unpack `ref`"),
+            }
+                .to_string()
```
Comment (marcusklaas) [inline src/message.rs:65]: This should be fixed in the latest rustfmt

File: src/config.rs [hunk 9]
```diff
@@ -524,7 +541,9 @@ mod tests {
               secret = "hi five"
             [app.playbooks]
               bad_path = "not-absolute"
-        "#.to_string()).unwrap();
+        "#
```
Comment (marcusklaas) [inline src/config.rs:544]: Oops...

File: src/config.rs (hunk 3 above)
Comment (marcusklaas) [inline src/config.rs:433]: That doesn't address this particular issue unfortunately :-(

**Comment by brianloveswords** [comment]
Is rustfmt making decisions based on an RFC? I'm wondering if some of the rules that led it to changes just indentation level could/should be implemented in https://github.com/rust-lang/rust-mode

File: src/repo_config.rs [hunk 10]
```diff
@@ -213,9 +211,9 @@ impl<'a> RepoConfig<'a> {
                 }),
                 LookupResult::Value(v) =>
                     match VerifiedPath::file(Some(project_root), Path::new(v)) {
-                        Ok(v) => Some(v),
-                        Err(err) => return Err(err),
-                    },
+                    Ok(v) => Some(v),
```
Comment (brianloveswords) [inline src/repo_config.rs:214]: @steveklabnik This one and the one below also look wrong to me

**Comment by brianloveswords** [comment]
I cherry-picked the parts I was sure made an improvement (most of the changes) but broke the merge by rearranging some `use` statements and hand-editing some other parts!

**Comment by nrc** [comment]
@brianloveswords there has been one RFC for function decls, but it was inconclusive. Rustfmt is heavily customisable at the moment. Once it is working really well (can be run reliably on the main repo, for example) and thus it is easy to do mass changes to formatting, I'll start more RFCs to nail down the style.

**Comment by brianloveswords** [comment]
Closing this because I pulled out the relevant stuff, thanks for pointing me towards rustfmt @steveklabnik (and thanks for making it @nrc!). I'll keep running it as I work on this and file any issues I see with the results :+1:

**Comment by steveklabnik** [comment]
:+1:

**Comment by jamesplease** [comment]
I read this as "rust, fml"

**Pull request closed without merging**